src/redprl/refiner_types.fun

```diff
@@ -708,30 +693,29 @@ struct
         val Syn.PATH_AP (m0, r0) = Syn.out ap0
         val Syn.PATH_AP (m1, r1) = Syn.out ap1
         val () = Assert.paramEq "Path.ApEq" (r0, r1)
-        val (goalSynth, holeSynth) = makeGoal @@ (I, H) >> CJ.SYNTH m0
-        val (goalMem, _) = makeGoal @@ (I, H) >> CJ.EQ ((m0, m1), holeSynth)
-        val (goalLine, holeLine) = makeGoal @@ MATCH (O.MONO O.PATH_TY, 0, holeSynth, [r0], [])
-        val (goalTy, _) = makeGoal @@ (I, H) >> CJ.EQ_TYPE (ty, holeLine)
+        val (goalSynth, holeSynth) = makeSynth (I, H) m0
+        val goalMem = makeEqIfDifferent (I, H) ((m0, m1), holeSynth) (* m0 well-typed *)
+        val (goalLine, holeLine) = makeMatch (O.MONO O.PATH_TY, 0, holeSynth, [r0], [])
+        val goalTy = makeEqType (I, H) (holeLine, ty) (* holeLine type? *)
```

Comment:
Yes, I think it should be possible to assume this as well, for the same reason as the other one.